feat: generate and upload a project's SBOM

patheard opened the pull request:
# Summary 
Add a shared GitHub workflow that can be used by downstream
projects to generate and upload SBOMs to our Dependency-Track
instance.

Currently supports creating CycloneDX SBOMs in `.json` format
for Node, PHP and Python projects.

# Related
* cds-snc/platform-sre-security-support#92
* cds-snc/platform-sre-security-support#94

patheard commented on terragrunt/aws/sso_proxy/container-definitions/pomerium_sso_proxy.json.tmpl:
```diff
@@ -1,65 +1,67 @@
+[
+  {
+    "name" : "pomerium_sso_proxy",
+    "cpu" : 0,
```
I don't think it's a big deal to have this, but do we need to set a 0 reserved CPU units for the container definition?  I'm used to letting our containers fight it out :)

mohamed-cds replied:
i misunderstood this previously and just happily copy pasted it around my projects. I've removed it now since i used to think it was needed to tell the container to use whatever is defined at the task def level.

patheard replied:
I had thought it was related to the task's `cpu` definition as well so had to look it up since I hadn't seen it on a container def before.

patheard committed:
feat: generate and upload a project's SBOM

Add a shared GitHub workflow that can be used by downstream
projects to generate and upload SBOMs to our Dependency-Track
instance.

Currently supports creating CycloneDX SBOMs in `.json` format
for Node, PHP and Python projects.

mohdnr reviewed (approved):
nice! i like how you found a way to have all the different generators in the same workflow. 🚀 ✅
